Default cover image for local feeds

damoasda opened the pull request:
Provide default cover image for local feeds

* The default cover image is used if the source folder doesn't contain a file like `folder.png` (see `LocalFeedUpdater`).
* Subscription list view: If the default cover image is shown for a feed, additionally show the feed name to keep it identifiable.
* Default cover image is in `res/raw/folder.png`

Contributes to #4287.

ByteHamster commented:
Does not seem to work on my phone. To make it work, you can change the following in `ApOkHttpUrlLoader`:

```java
    @Override
    public boolean handles(@NonNull String model) {
        // Leave some URIs to Glide's default loaders
        return !TextUtils.isEmpty(model)
                && !model.startsWith(ContentResolver.SCHEME_CONTENT)
                && !model.startsWith(ContentResolver.SCHEME_ANDROID_RESOURCE);
    }
```

I really like the fact that the title is still displayed on the subscriptions screen, on top of the folder image.

I would add a bit more padding to the image and make the background darker. Currently, the background is just a tiny bit darker than the navigation drawer background, which looks a bit odd to me. While thinking about this, I noticed that the folder icon might conflict with #1711. Maybe it would be better to come up with a slightly different icon. Any ideas? Maybe using a white folder on a blue background could work.

The text on the subscriptions screen when (normal) feeds do not have a cover has pretty low contrast. This was fixed in #4243. I guess we need to merge `develop` into `add-local-feeds` soon. I will do that after this PR is merged. Before that, please make sure that your PR still looks good with those changes merged.

We need to make sure that the file is not deleted by proguard when doing a release build. I have not tried it yet but I thing that this might happen. Additionally, I think that Android Studio ("Find usages" or just inspections) will also think that the file is unused. If you use something like `getResources().getResourceEntryName(R.raw.folder)` instead of `"raw/folder"`, both problems should be solved at once. I would change the name of the icon to be a bit more descriptive. Something like `local_feed_default_icon`.

keunes commented:
> While thinking about this, I noticed that the folder icon might conflict with #1711. Maybe it would be better to come up with a slightly different icon. Any ideas? Maybe using a white folder on a blue background could work.

👍

ByteHamster commented:
I'm just wondering if a *file* icon instead of a *folder* icon would be even less ambiguous:

![file-outline](https://user-images.githubusercontent.com/5811634/88486235-9d4a2d80-cf7c-11ea-84c0-896c06c6e9cb.png)

keunes commented:
> I'm just wondering if a file icon instead of a folder icon would be even less ambiguous:

This feature, if I understand correctly, scans a folder for all files in it. The feed represents then the folder, the episode the files. So I would say no (to it being less ambiguous). I think white on a blue background would work nice.

damoasda commented:
@ByteHamster: Thank you for the detailed feedback and the hints. I did some code changes. See my comments below.

> Does not seem to work on my phone. To make it work, you can change the following in `ApOkHttpUrlLoader`

Thank you for the hint, this solved strange behaviour also on my side. I tried a lot of code variations until all icons worked for me, but it must have been only caching effects.

> I would add a bit more padding to the image and make the background darker.

I replaced the image by a new one, a white folder icon on blue background. It looks a lot better now. Please let me know if you want more changes. I can also provide the corresponding SVG file if you are interested.

> The text on the subscriptions screen when (normal) feeds do not have a cover has pretty low contrast. This was fixed in #4243. I guess we need to merge `develop` into `add-local-feeds` soon. I will do that after this PR is merged. Before that, please make sure that your PR still looks good with those changes merged.

I adopted the changes in #4243. Unfortunately, I had to use a different background color if both, the cover image and the feed name, are shown at the same time. See `CoverLoader.CoverTarget.onResourceReady()` for details. Not an optimal solution in my opinion, maybe somebody has a better idea?

> We need to make sure that the file is not deleted by proguard when doing a release build. I have not tried it yet but I thing that this might happen. Additionally, I think that Android Studio ("Find usages" or just inspections) will also think that the file is unused. If you use something like `getResources().getResourceEntryName(R.raw.folder)` instead of `"raw/folder"`, both problems should be solved at once. I would change the name of the icon to be a bit more descriptive. Something like `local_feed_default_icon`.

You were right, Android Studio didn't detect the usage of the file. The file is referred by `getResourceEntryName()` now. I also renamed the file as suggested.

@keunes: The old file was just a screenshot of `ic_folder_black.xml` in Android Studio. The new file is based on the icons `folder` and `audiotrack` in Android Studio that I combined with Inkscape to a SVG file and finally exported to PNG. So in terms of licenses it should be fine.

ByteHamster commented:
> Please let me know if you want more changes. I can also provide the corresponding SVG file if you are interested.

AntennaPod uses vector drawables for most other icons. Is is possible to load vector drawables with Glide? That would make the icon more crisp and the app smaller.

> So in terms of licenses it should be fine.

:+1:

damoasda commented:
> AntennaPod uses vector drawables for most other icons. Is is possible to load vector drawables with Glide? That would make the icon more crisp and the app smaller.

I already tried to make it work with the Android Drawable format (xml) but I haven't been successful.

ByteHamster commented:
> I already tried to make it work with the Android Drawable format (xml) but I haven't been successful.

Okay, then it's fine for now. Thank you very much for working on local feed support!
